sources/get_list_source.go

```diff
@@ -42,11 +42,26 @@ type GetListSource[AWSItem AWSItemType, ClientStruct ClientStructType, Options O
 	// ItemMapper Maps an AWS representation of an item to the SDP version
 	ItemMapper func(scope string, awsItem AWSItem) (*sdp.Item, error)
 
+	// ItemMapperWithQuery Maps an AWS representation of an item to the SDP version
+	ItemMapperWithQuery func(query, scope string, awsItem AWSItem) (*sdp.Item, error)
```

Comment:
I can't find where we're actually using this, am I missing it somewhere?

Reply:
Please see this comment: https://github.com/overmindtech/aws-source/pull/608#discussion_r1751010177